BasalDose/BasalDose.c

```diff
@@ -70,37 +49,16 @@ void TIMER0_IRQHandler(void)
 	if(StepperMotor_GlobalPosition + BASAL_STEPS <= SYRINGE_LENGTH)
 	{
 		Control_GlobalStatus = Basal;
-		LED_On(0); // Signal that Basal is being administered P1.28
+		Control_GlobalState = Administration;
+		LPC_GPIO1->FIOSET |= 1 << 28; // Signal that Basal is being administered P1.28
 	}
 	else
 	{
-		Control_GlobalStatus = Backward;
-		LED_On(2); // Signal that Backward/Retraction is occuring P1.31
+		Control_GlobalStatus = None;
+		Control_GlobalState = Full;
```

Comment:
At this point, it would mean that the syringe did not have enough insulin to complete a dosage. So the state should either be empty, or an intermediate state that represents that we need to check while administering whether or not the syringe is empty.